code/modules/mob/living/simple_animal/friendly/farm_animals.dm

```diff
@@ -155,6 +155,72 @@
 	else
 		..()
 
+/mob/living/simple_animal/friendly/sheep
+	name = "sheep"
+	desc = "It's so fluffy! can't shear it though. milking is fine."
+	icon_state = "sheep"
+	icon_living = "sheep"
+	icon_dead = "sheep_dead"
+	icon_gib = "cow_gib"
+	speak_emote = list("baas","baas hauntingly")
+	turns_per_move = 5
+	see_in_dark = 6
+	response_help  = "pets"
+	response_disarm = "gently pushes aside"
+	response_harm   = "kicks"
+	health = 50
+
+	meat_type = /obj/item/reagent_containers/food/snacks/meat
+	meat_amount = 4
+	bone_amount = 6
+	skin_material = MATERIAL_SKIN_COWHIDE
+	skin_amount = 4
+
+	var/datum/reagents/udder = null
+
+	ai_holder_type = /datum/ai_holder/simple_animal/passive/cow
+	say_list_type = /datum/say_list/cow
+
+/mob/living/simple_animal/friendly/sheep/New()
+	udder = new(50, src)
+	..()
+
+/mob/living/simple_animal/friendly/sheep/attackby(var/obj/item/O as obj, var/mob/user as mob)
+	var/obj/item/reagent_containers/glass/G = O
+	if(stat == CONSCIOUS && istype(G) && G.is_open_container())
+		user.visible_message("<span class='notice'>[user] milks [src] using \the [O].</span>")
+		var/transfered = udder.trans_type_to(G, /datum/reagent/drink/milk, rand(5,10))
+		if(G.reagents.total_volume >= G.volume)
+			to_chat(user, "<span class='warning'>\The [O] is full.</span>")
+		if(!transfered)
+			to_chat(user, "<span class='warning'>The udder is dry. Wait a bit longer...</span>")
+	else
+		..()
+
+/mob/living/simple_animal/friendly/sheep/Life()
+	. = ..()
+	if(!.)
+		return FALSE
+	if(udder && prob(5))
+		udder.add_reagent(/datum/reagent/drink/milk/sheep, rand(5, 10))
+
+/mob/living/simple_animal/friendly/sheep/attack_hand(mob/living/carbon/M as mob)
+	if(!stat && M.a_intent == I_DISARM && icon_state != icon_dead)
+		M.visible_message("<span class='warning'>[M] tips over [src].</span>","<span class='notice'>You tip over [src].</span>")
+		Weaken(30)
+		icon_state = icon_dead
+		spawn(rand(20,50))
```

Comment:
Spawns are yuuuuuuucky. Absolutely look into making an `addtimer` for this instead to call a separate proc.